### DISCUSSION TITLE
Question about how "server_names" works

### DISCUSSION THREAD

**Discussion by seth100** [DNSCrypt/dnscrypt-proxy#2437] [Q&A] [answered]
Hi,
I have a doubt about how `server_names` works:

I tried to comment the `server_names` property so that dnscrypt chooses the fastest servers for me and I see from dnsleaktest.com that my system is connecting to many servers so I guess it means that dnscrypt is using **all** the available servers (based of course on my filter preferences, I see there are 32 live ATM) and so every time I make a DNS request my system connects to **all** those servers, not to just the fastest one, is that right?
If so, isn't that a slowdown for my connection requests?

Thanks

**Comment by lifenjoiner** [comment, marked as answer]
No, in short.

1. By default, it picks one following `lb_strategy`.
2. The servers delay are measured by `EWMA` algorithm on the fly, and ranked dynamically.
3. Those testing sites always try a bunch of specified domains. If delay of the last answer goes high, its ranking is likely to be dropped, and then a "better" server will be adopted.

In summary, it always tries to adopt the best or one of the best N servers, on the fly.

**Reply by seth100** [reply]
Nice, thanks for answering!
So why do I see many servers in dnsleaktest.com?

**Reply by lifenjoiner** [reply]
The servers ranking changed during solving those specified domains.

**Reply by seth100** [reply]
oh ok, cool thanks!
